code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -1,604 +1,618 @@
+			src << alert("The arrivals shuttle is currently malfunctioning! You cannot join.")
+			return FALSE
+
+		if(CONFIG_GET(flag/arrivals_shuttle_require_undocked))
+			SSshuttle.arrivals.RequireUndocked(src)
+		arrivals_docked = SSshuttle.arrivals.mode != SHUTTLE_CALL
+
+	//Remove the player from the join queue if he was in one and reset the timer
+	SSticker.queued_players -= src
+	SSticker.queue_delay = 4
+
+	SSjob.AssignRole(src, rank, 1)
+
+	var/mob/living/character = create_character(TRUE)	//creates the human and transfers vars and mind
+	var/equip = SSjob.EquipRank(character, rank, TRUE)
+	if(isliving(equip))	//Borgs get borged in the equip, so we need to make sure we handle the new mob.
+		character = equip
+
+	var/datum/job/job = SSjob.GetJob(rank)
+
+	if(job && !job.override_latejoin_spawn(character))
+		SSjob.SendToLateJoin(character)
+		if(!arrivals_docked)
+			var/obj/screen/splash/Spl = new(character.client, TRUE)
+			Spl.Fade(TRUE)
+			character.playsound_local(get_turf(character), 'sound/voice/ApproachingTG.ogg', 25)
+
+		character.update_parallax_teleport()
+
+	SSticker.minds += character.mind
+
+	var/mob/living/carbon/human/humanc
+	if(ishuman(character))
+		humanc = character	//Let's retypecast the var to be human,
+
+	if(humanc)	//These procs all expect humans
+		GLOB.data_core.manifest_inject(humanc)
+		if(SSshuttle.arrivals)
+			SSshuttle.arrivals.QueueAnnounce(humanc, rank)
+		else
+			AnnounceArrival(humanc, rank)
+		AddEmploymentContract(humanc)
+		if(GLOB.highlander)
+			to_chat(humanc, "<span class='userdanger'><i>THERE CAN BE ONLY ONE!!!</i></span>")
+			humanc.make_scottish()
+
+		if(GLOB.summon_guns_triggered)
+			give_guns(humanc)
+		if(GLOB.summon_magic_triggered)
+			give_magic(humanc)
+
+	GLOB.joined_player_list += character.ckey
+	GLOB.latejoiners += character
+
+	if(CONFIG_GET(flag/allow_latejoin_antagonists) && humanc)	//Borgs aren't allowed to be antags. Will need to be tweaked if we get true latejoin ais.
+		if(SSshuttle.emergency)
+			switch(SSshuttle.emergency.mode)
+				if(SHUTTLE_RECALL, SHUTTLE_IDLE)
+					SSticker.mode.make_antag_chance(humanc)
+				if(SHUTTLE_CALL)
+					if(SSshuttle.emergency.timeLeft(1) > initial(SSshuttle.emergencyCallTime)*0.5)
+						SSticker.mode.make_antag_chance(humanc)
+
+	if(humanc && CONFIG_GET(flag/roundstart_traits))
+		SSquirks.AssignQuirks(humanc, humanc.client, TRUE)
+
+	log_manifest(character.mind.key,character.mind,character,latejoin = TRUE)
+
+/mob/dead/new_player/proc/AddEmploymentContract(mob/living/carbon/human/employee)
+	//TODO:  figure out a way to exclude wizards/nukeops/demons from this.
+	for(var/C in GLOB.employmentCabinets)
+		var/obj/structure/filingcabinet/employment/employmentCabinet = C
+		if(!employmentCabinet.virgin)
+			employmentCabinet.addFile(employee)
+
+
+/mob/dead/new_player/proc/LateChoices()
+
+	var/level = "green"
+	switch(GLOB.security_level)
+		if(SEC_LEVEL_GREEN)
```

Comment:
@BurgerLUA I added this because it was silly not to have a switch if someone went from blue to green.